docs/source/plugin/advanced.rst

```diff
@@ -64,3 +64,115 @@ a QUIT/PART event, and they will be available after a JOIN event.
 
 Note that you don't specifically need to use ``@plugin.thread(False)``, but
 it is still recommended to prevent any race condition.
+
+
+Capability Negociation
+======================
+
+Capability negociation is a feature of IRCv3 that allows a server to advertise
+a list of optional capabilities, and allows its clients to request such
+capabilities. You can see that as feature flags, activated by the client.
+
+Capability negociation takes place after:
+
+* connecting to the IRC server
+* client's identification (``USER`` and ``NICK``)
+
+And before:
+
+* the ``RPL_WELCOME`` event (001)
+* ``ISUPPORT`` messages
+* client's authentification (except for SASL, which occurs in the capability
+  negociation phase)
+
+.. warning::
+
+    This is a very advanced feature, and plugin authors should understand how
+    capability negociation works before using it. Even if Sopel tries to make
+    it as simple as possible, plugin authors should be aware of the known
+    limitations and possible caveats.
+
+Declaring requests: the ``capability`` decorator
+------------------------------------------------
+
+In :mod:`sopel.plugin` there is an advanced :class:``capability`` decorator: it
+is a class that acts like a decorator, and can be used with or without a
+callback to decorate::
+
+    # plugin file
+    from sopel import plugin
+
+    # this will register a capability request
+    CAP_ACCOUNT_TAG = plugin.capability('account-tag')
+
+    # this will work as well
+    @plugin.capability('message-prefix')
+    def cap_message_prefix(cap_req, bot, acknowledged):
+        # do something if message-prefix is ACK or NAK
+        ...
+
+.. autoclass:: sopel.plugin.capability
+   :members:
+
+.. autoclass:: sopel.plugin.CapabilityNegotiation
+   :members:
+
+
+Working with capabilities
+-------------------------
+
+A plugin that requires capabilities, or that can enhance its features with
+capabilities, should rely on the bot's
+:attr:`capability manager <sopel.irc.AbstractBot.capabilities>`::
```

Comment:
Not strictly related to this PR, but contextually relevant: Did we ever figure out how to make x-refs like this appear in not-monospace? When the link text is overridden to something that's not a code object, it looks odd.

Reply:
Oh. I think there is a trick, I'll see what I can do to make it less odd. The minimum I can do is to replace this by `bot.capabilities` to make it less odd.